### DISCUSSION TITLE
What does `groupbyx` create?

### DISCUSSION THREAD

**Discussion by mo-joshuacolclough** [hosseinmoein/DataFrame#256] [Q&A] [answered]
I see in the [documentation](https://htmlpreview.github.io/?https://github.com/hosseinmoein/DataFrame/blob/master/docs/HTML/groupby.html) that the `groupbyx` function returns a `DataFrame` object.
Is this a "view" of the original DataFrame? Or is this a separate, new copy containing only the grouped items. I.e, if I was to group elements in a `first` DataFrame, and edit the resulting `second` DataFrame, would making changes in `second` also make changes in the `first` DataFrame?

As a seperate but somewhat related question, if I am to then use `get_column` on this `second` DataFrame, is it a reference to a vector of values? Or a reference to a vector of pointers? I'm looking at operating on these objects in Fotran (via reference) - it's important to know that it's changing the original data. Thanks!

**Comment by hosseinmoein** [comment, marked as answer]
All groupby's return a seperate brand new DataFrame. That should be evident from the documentation. 
When you call get_column on a DataFrame  (as opposed to a view) you get a reference to a vector of values.

**Reply by mo-joshuacolclough** [reply]
OK great, thanks
